src/main/java/com/crud/cadastrousuario/domain/service/CrudUserService.java

```diff
@@ -0,0 +1,95 @@
+package com.crud.cadastrousuario.domain.service;
+
+
+import com.crud.cadastrousuario.domain.dto.UserDTO;
+import com.crud.cadastrousuario.domain.exception.BadRequestException;
+import com.crud.cadastrousuario.domain.exception.NotFoundException;
+import com.crud.cadastrousuario.domain.model.User;
+import com.crud.cadastrousuario.domain.repository.UserRepository;
+import com.crud.cadastrousuario.domain.repository.UserRepositorySpec;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.stereotype.Service;
+import org.springframework.data.domain.Pageable;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class CrudUserService {
+
+    @Autowired
+    public UserRepository userRepository;
+
+    public User save(User user) throws BadRequestException {
+
+        isEmailAvailable(user);
+        return userRepository.save(user);
+    }
+
+    public List<User> findUser(Pageable pageable , UserDTO filter){
+
+        Page<User> pageUser = userRepository.findAll(
+                UserRepositorySpec.filter(filter),
+                PageRequest.of(pageable.getPageNumber(),
+                        pageable.getPageSize()));
+
+        return pageUser.getContent();
+
+    }
+
+    public User findUserByID(Long id) throws NotFoundException {
+        isIdAvailable(id);
+        Optional<User> opt = userRepository.findById(id);
```

Comment:
quando recuperar a informação do user , é melhor pegar o resulta e colocar um optional.isEmpty para validar e lançar a exceção not found